src/main/java/com/aws/greengrass/smbridge/clients/SMClient.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.smbridge.clients;
+
+import com.amazonaws.greengrass.streammanager.client.StreamManagerClientFactory;
+import com.amazonaws.greengrass.streammanager.client.exception.StreamManagerException;
+import com.amazonaws.greengrass.streammanager.model.MessageStreamDefinition;
+import com.amazonaws.greengrass.streammanager.model.MessageStreamInfo;
+import com.amazonaws.greengrass.streammanager.model.StrategyOnFull;
+import com.amazonaws.greengrass.streammanager.model.export.ExportDefinition;
+import com.aws.greengrass.componentmanager.KernelConfigResolver;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.smbridge.StreamMessage;
+import com.aws.greengrass.smbridge.StreamDefinition;
+import com.amazonaws.greengrass.streammanager.client.StreamManagerClient;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class SMClient {
+    private static final Logger LOGGER = LogManager.getLogger(SMClient.class);
+
+    private StreamManagerClient smClient;
+    private MessageStreamDefinition defaultStream;
+    /**
+     * Ctr for SMClient.
+     *
+     * @param topics             topics passed in by Nucleus
+     * @throws SMClientException if unable to create SM Client
+     */
+    @Inject
+    public SMClient(Topics topics) throws SMClientException {
+        this(topics, null);
+        // TODO: Handle the case when serverUri is modified
+        try {
+            this.smClient = StreamManagerClientFactory.standard().build();
+        } catch (StreamManagerException e) {
+            throw new SMClientException("Unable to create a SM client", e);
+        }
+    }
+    protected SMClient(Topics topics, StreamManagerClient streamManagerClient){
+        this.smClient = streamManagerClient;
+        // TODO: Configurable default stream
+        this.defaultStream = new MessageStreamDefinition();
```

Comment:
What is the purpose of the default stream?

Reply:
If a message is received on one of the "reserved" MQTT topics, then it will be sent to this default stream, since the user has not configured any other stream for it to go to.

Reply:
What are the "reserved" topic? Example?

Reply:
The team would decide. Was going to sync with you and Joe when I get to implementing that logic. This is for the no-config functionality that GG would define some reserved topics that the user can use to automatically ingest MQTT -> SM with no config of the bridge.

Reply:
Example off the top of my head could be greengrass/reserved/streammanager

Reply:
We can talk about it, I was imagining that the name of the stream would be in the topic itself.

Reply:
That's an easy code change. I can push a new commit?

Reply:
I think it needs a discussion. I don't want to block this change. Lets talk about this offline.